packages/core/src/pipeline/details/StageFailureMessage.tsx

```diff
@@ -71,18 +71,30 @@ export class StageFailureMessage extends React.Component<IStageFailureMessagePro
   }
 
   public render() {
-    const { message, messages } = this.props;
+    const { message, messages, stage } = this.props;
     const { isFailed, failedTask, failedExecutionId, failedStageName, failedStageId } = this.state;
-    if (isFailed || failedTask || message || messages.length) {
+
+    const stageMessages = message || !messages.length ? [message] : messages;
+    if (stageMessages.length > 0) {
       const exceptionTitle = isFailed ? (messages.length ? 'Exceptions' : 'Exception') : 'Warning';
-      const displayMessages =
-        message || !messages.length ? (
-          <Markdown message={message || StageFailureMessages.NO_REASON_PROVIDED} className="break-word" />
-        ) : (
-          messages.map((m, i) => (
-            <Markdown key={i} message={m || StageFailureMessages.NO_REASON_PROVIDED} className="break-word" />
-          ))
-        );
+
+      let filteredMessages: string[] = [];
+      if (isFailed || stage.context?.skipExpressionEvaluation) {
+        // filter out expression evaluation failure messages if:
+        // - there was an actual failure as these can get *really* long and hide the failure message
+        // - expression evaluation was explicitly disabled (as Orca still processes these and populates errors
+        //   even when disabled)
+        filteredMessages = stageMessages.filter((m) => !m.startsWith('Failed to evaluate'));
+      }
```

Comment:
Can you explain this further?  Why would we filter "failed to evaluate" messages when "isFailed" is true and "skipExpressionEvaluation" is false?

Reply:
The failed to evaluate messages can get extremely long and are displayed even if expression evaluation is explicitly skipped in the stage config. In contrast the failure message is usually quite small. So if there's an actual failure message, OR if skipExpressionEvaluation is true, filtering them out removes noise and lets you see the actual error.

An example of this would be where an entire configmap is being printed because Orca failed to evaluate expressions in it, *even though* expression evaluation was explicitly disabled in the stage.